superset/databases/api.py

```diff
@@ -335,3 +351,88 @@ def select_star(
         return self.response(200, result=result)
+
+    @expose("/test_connection", methods=["POST"])
+    @protect()
+    @safe
+    @event_logger.log_this
+    @statsd_metrics
+    def test_connection(  # pylint: disable=too-many-return-statements
+        self,
+    ) -> FlaskResponse:
+        """Tests a database connection
+        ---
+        post:
+          description: >-
+            Tests a database connection
+          requestBody:
+            description: Database schema
+            required: true
+            content:
+              application/json:
+                schema:
+                  type: object
+                  properties:
+                    encrypted_extra:
+                      type: object
+                    extras:
+                      type: object
+                    name:
+                      type: string
+                    server_cert:
+                      type: string
+          responses:
+            200:
+              description: Database Test Connection
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      message:
+                        type: string
+            400:
+              $ref: '#/components/responses/400'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        uri = request.json.get("uri")
+        try:
+            DatabaseDAO.test_connection(
+                db_name=request.json.get("name"),
+                uri=uri,
+                server_cert=request.json.get("server_cert"),
+                extra=json.dumps(request.json.get("extras", {})),
+                impersonate_user=request.json.get("impersonate_user"),
+                encrypted_extra=json.dumps(request.json.get("encrypted_extra", {})),
+            )
+            return self.response(200, message="OK")
+        except CertificateException as ex:
+            logger.info("Certificate exception")
+            return self.response(500, message=ex.message)
+        except (NoSuchModuleError, ModuleNotFoundError):
+            logger.info("Invalid driver")
+            driver_name = make_url(uri).drivername
+            message = "Could not load database driver: %s" % (driver_name)
+            return self.response(400, message=message, driver_name=driver_name)
+        except ArgumentError:
+            logger.info("Invalid URI")
+            return self.response_422(
+                message="Invalid connection string, a valid string usually follows:\n"
+                "'DRIVER://USER:PASSWORD@DB-HOST/DATABASE-NAME'"
+            )
+        except OperationalError:
+            logger.warning("Connection failed")
+            return self.response(
+                500, message="Connection failed, please check your connection settings"
```

Comment:
I think we're missing the `_()` function that will hook these messages up with their associated translations.